docs/source/nested.rst

```diff
@@ -39,6 +39,20 @@ nested_tensor([
   tensor([3., 4., 5., 6., 7.], device='cuda:0')
 ])
 
+Note that the passed Tensors are being copied into a contiguous piece of memory. The resulting
+NestedTensor allocates new memory to store them and does not keep a reference.
+
+In order to form a valid NestedTensor the passed Tensors also all need to match in dimension.
```

Comment:
"match in dimension" => meaning in number of dimensions? or in dimension values?

it's not very clear what this means... also not very clear currently if differently-sized images can be put inside? e.g. two tensors of shapes: [(3, 240, 320), (3, 640, 480)].

It would be very useful to explain concrete thought-of usage examples (e.g. variable-sized BTC (?) tensors) and some concrete example (images in vision) that don't work for now. I'm mentioning this e.g. because in DETR codebase a similar NestedTensor name was used to mean TensorList with two variable spatial dimensions.

Reply:
Ah, good point! Thanks for the review!

I wrote a lot about this in [pytorch/nestedtensor](https://colab.research.google.com/github/pytorch/nestedtensor/blob/master/tutorials/notebooks/basic.ipynb#scrollTo=LwZM_uuaW_Cg), but not all features required for this tutorial have been ported yet. We could think about setting that as a goal. What do you think?

Reply:
Well, I'm just worried that some existing design (with goal for transformers / NLP) gets baked in, and the rest is left forgotten. So yes, I think it's best to move at least some of this thoughts at very visible public docs, so users can request this stuff...

Reply:
@vadimkantorov - I'm very keen on preventing that from happening and this is a good point. NestedTensors reach far beyond the NLP + Transformers use case.